mesonbuild/compilers/compilers.py

```diff
@@ -282,10 +282,12 @@ def are_asserts_disabled(target: 'BuildTarget', env: 'Environment') -> bool:
             (env.coredata.get_option_for_target(target, 'b_ndebug') == 'if-release' and
              env.coredata.get_option_for_target(target, 'buildtype') in {'release', 'plain'}))
 
+
 def are_asserts_disabled_for_subproject(subproject: str, env: 'Environment') -> bool:
-    return (env.coredata.get_option_for_subproject('b_ndebug', subproject) == 'true' or
-            (env.coredata.get_option_for_subproject('b_ndebug', subproject) == 'if-release' and
-             env.coredata.get_option_for_subproject('buildtype', subproject) in {'release', 'plain'}))
+    key = OptionKey('b_ndebug', subproject)
+    return (env.coredata.optstore.get_value_for(key) == 'true' or
+            (env.coredata.optstore.get_value_for(key.evolve(name='b_ndebug')) == 'if-release' and
+             env.coredata.optstore.get_value_for(key.evolve(name='buildtype')) in {'release', 'plain'}))
```

Comment:
evolve is missing the subproject!

Reply:
It doesn't need it? the whole point of `.evolve()` is that only the named parameters are changed and the rest are copied from the original.

Reply:
Ah, yes it is.